### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by Marvin-Beckmann**
**Description**

This repository has not been refactored yet, and especially in terms of interfaces, we were able to have much nicer interfaces. This PR shall optimize our code base and make it more consistent.

- [x] Use Into<...> wherever it makes sense for initialization, e.g. signatures
- [x] Conventions doc comments test functions
- [x] Useless imports in doc comments
- [x] Reduce complexity and length of doc comments
- [x] Fix Typos
- [x] Use math notation consistently in the crate
- [x] Restructure crate s.t. sampling from distributions, etc. is deleted (i.e. anywhere where our plans regarding implementation crate changed)

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases
- [x] I provided an intuition regarding how certain inputs have to be set

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/construction/signature/fdh.rs [hunk 1]
```diff
@@ -39,26 +39,17 @@ pub mod serialize;
...
-/// let mut fdh = Fdh::init_gpv(n, &modulus, &s);
+/// let mut fdh = Fdh::init_gpv(4, 113, 17);
 ///
 /// let m = "Hello World!";
 ///
 /// let (pk, sk) = fdh.gen();
 /// let sigma = fdh.sign(m.to_owned(), &sk, &pk);
 ///
-/// assert_eq!(&sigma, &fdh.sign(m.to_owned(), &sk, &pk));
 /// assert!(fdh.vfy(m.to_owned(), &sigma, &pk))
```
Comment (Marcel583) [inline src/construction/signature/fdh.rs:53]:
```suggestion
/// assert!(fdh.vfy(m.to_owned(), &sigma, &pk));
```

File: src/construction/signature/fdh/gpv.rs [hunk 2]
```diff
@@ -37,40 +37,34 @@ impl Fdh<MatZq, (MatZ, MatQ), MatZ, MatZq, PSFGPV, HashMatZq> {
...
     ///
+    /// let mut fdh = Fdh::init_gpv(4, 113, 17);
     /// let (pk, sk) = fdh.gen();
-    /// let sigma = fdh.sign(m.to_owned(), &sk, &pk);
     ///
-    /// assert_eq!(&sigma, &fdh.sign(m.to_owned(), &sk, &pk));
-    /// assert!(fdh.vfy(m.to_owned(), &sigma, &pk))
+    /// let sigma = fdh.sign(m.to_string(), &sk, &pk);
+    ///
+    /// assert!(fdh.vfy(m.to_string(), &sigma, &pk))
```
Comment (Marcel583) [inline src/construction/signature/fdh/gpv.rs:49]:
```suggestion
    /// assert!(fdh.vfy(m.to_string(), &sigma, &pk));
```

File: src/construction/signature/fdh/serialize.rs [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 //! Allows to Deserialize an arbitrary [`Fdh`] instantiation
```
Comment (Marcel583) [inline src/construction/signature/fdh/serialize.rs:9]:
```suggestion
//! Allows to Deserialize an arbitrary [`Fdh`] instantiation;
```

File: src/construction/signature/pfdh/gpv.rs [hunk 4]
```diff
@@ -50,6 +49,9 @@ impl Pfdh<MatZq, (MatZ, MatQ), MatZ, MatZq, PSFGPV, HashMatZq> {
     ///
     /// assert!(pfdh.vfy(m.to_owned(), &sigma, &pk))
```
Comment (Marcel583) [inline src/construction/signature/pfdh/gpv.rs:50]:
```suggestion
    /// assert!(pfdh.vfy(m.to_owned(), &sigma, &pk));
```

**Review by Marcel583 — APPROVED** [review]
